Update mcscf fix_spin_ user doc (#76)
pyscf/pyscf.github.io: source/user/mcscf.rst

  The `frozen` keyword of the CASSCF optimizer should not be confused with the `frozen` keyword of the FCI solver, which controls the number of orbitals that are constrained to be doubly occupied.

Spin state of CAS wavefunction
------------------------------

> The MCSCF wavefunction by default has the same Sz value as the setting of
> :attr:`mol.spin`. The program does not guarantee to derive the spin-adapted wave
> function of desired spin multiplicity. There are a few settings can be used
> to tune the spin multiplicity of the wavefunction.

Reviewer: ```suggestion
By default, the MCSCF wavefunction uses the setting in :attr:`mol.spin` by default for the value of Sz. 
However, there is no guarantee that a spin-adapted wave function of the desired spin multiplicity is obtained. 
There are a few settings that can be used to tune the spin multiplicity of the wavefunction.
```

> You can change the number of alpha and beta electrons to be correlated in active
> space. The Sz value of the MCSCF wavefunction can be different to the settings
> of :attr:`mol.spin`. It's common to start from `Sz=0` RHF calculations then
> setting different number of alpha and beta electrons in active space to solve triplet
> state. For example

Reviewer: ```suggestion
You can change the number of alpha and beta electrons in the active space.
The Sz value of the MCSCF wavefunction can differ from :attr:`mol.spin`.
It's common to start from `Sz=0` RHF calculations, and then to vary the 
number of alpha and beta electrons in the active space to solve higher 
spin states, such as the triplet state. For example```

.. code-block:: python

  mol.spin = 0
  myhf = mol.RHF().run()
  # 5 alpha electrons, 3 beta electrons
  mycas = mcscf.CASSCF(myhf, 6, (5, 3))
  mycas.kernel()

> It happens that the program give a wave-function of correct Sz while the spin
> multiplicity is wrong (or even a spin-contaminated wavefunction). The
> CASCI/CASSCF class provided the :func:`fix_spin_` method to correct the spin state.

Reviewer: ```suggestion
It may happen that a wave function of correct Sz is achieved while
S^2 is wrong. This issue may either be caused by convergence onto
another spin state, or spin contamination in the CAS wave function.

Such issues can be circumvented with the :func:`fix_spin_` method
of the CASCI/CASSCF class, which is able to correct the spin state
by biasing the calculation towards the wanted state.
```

.. code-block:: python

  mol.spin = 0
  myhf = mol.RHF().run()
  # 5 alpha electrons, 3 beta electrons
  mycas = mcscf.CASSCF(myhf, 6, (4, 4))
  # Targeting triplet state
  mycas.fix_spin_(ss=2)
  mycas.kernel()

> :func:`fix_spin_` is an energy-penalty method. It may affect the efficiency and
> accuracy of CASCI and CASSCF algorithm. It should not be used unless
> the MCSCF program get difficulty to find the correct spin state.

Reviewer: ```suggestion
:func:`fix_spin_` is an energy-penalty method. It may affect the efficiency and
accuracy of the CASCI and CASSCF algorithms. It should not be used unless
the MCSCF program has difficulties in finding the wanted spin state.
```

  The `frozen` keyword of the CASSCF optimizer should not be confused with the `frozen` keyword of the FCI solver, which controls the number of orbitals that are constrained to be doubly occupied.

Spin state of CAS wavefunction
------------------------------

> By default, the MCSCF wavefunction uses the setting in :attr:`mol.spin` by default for the value of Sz. 

Reviewer 2: I would add an introductory sentence explaining the spin-adaptation vs. spin projection. Something like "Multiconfigurational wavefunction provided by the the `pyscf.fci` solver is (typically?) spin-adapted, but there is no direct control of the spin multiplicity (i.e. S^2 value). It is, nevertheless, possible to define the spin projection Sz of the obtained WF, which helps to fix the spin multiplicity in most of the cases. By default, the MCSCF uses the setting in :attr:`mol.spin` for the value of 2*Sz."

> by default

was repeated two times in one line

Author: Updated

However, there is no guarantee that a spin-adapted wave function of the desired spin multiplicity is obtained. 
There are a few settings that can be used to tune the spin multiplicity of the wavefunction.

You can change the number of alpha and beta electrons in the active space.
The Sz value of the MCSCF wavefunction can differ from :attr:`mol.spin`.

> It's common to start from `Sz=0` RHF calculations, and then to vary the 

Reviewer 2: For transition-metal system with open *d* shell I would actually start with single-reference `Sz=max` state and then switch to more complicated low-spin states in CASSCF.

Author: Updated

number of alpha and beta electrons in the active space to solve higher 
spin states, such as the triplet state. For example```

.. code-block:: python

  mol.spin = 0
  myhf = mol.RHF().run()
  # 5 alpha electrons, 3 beta electrons
  mycas = mcscf.CASSCF(myhf, 6, (5, 3))
  mycas.kernel()

It may happen that a wave function of correct Sz is achieved while
S^2 is wrong. This issue may either be caused by convergence onto
another spin state, or spin contamination in the CAS wave function.

Such issues can be circumvented with the :func:`fix_spin_` method
of the CASCI/CASSCF class, which is able to correct the spin state
by biasing the calculation towards the wanted state.

.. code-block:: python

  mol.spin = 0
  myhf = mol.RHF().run()
  # 5 alpha electrons, 3 beta electrons

>   mycas = mcscf.CASSCF(myhf, 6, (4, 4))

Reviewer 2: `6, (4,4)`  contradicts the comment above

Author: Fixed

  # Targeting triplet state
  mycas.fix_spin_(ss=2)
  mycas.kernel()

:func:`fix_spin_` is an energy-penalty method. It may affect the efficiency and

> accuracy of the CASCI and CASSCF algorithms. It should not be used unless

Reviewer 2: Just a comment: isn't it a good idea to enable it automatically if the obtained S^2 is different from the Sz*(Sz+1)?

Author: For the moment I tend to keep the default fci solver for mcscf the simplest one. I suspect this treatment can cause numerical instability for complicated systems. In the future it should be possible to make the mcscf object more smart to automatically set the spin treatment.
